File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    today = new Date(today);
    var idx = 1;
    for(let privacy of privacies){
        var writeDate = new Date(privacy.split(" ")[0]);
        let writeTerm = privacy.split(" ")[1];
        var endDate = writeDate;
        
        for(let term of terms){
            if(term.split(" ")[0] == writeTerm){
                let whileMonth = parseInt(term.split(" ")[1]);
                endDate = new Date(endDate.setMonth(endDate.getMonth() + whileMonth));
            }
        }
        console.log(endDate, today)
        if(endDate <= today){
            answer.push(idx);
        }
        idx ++;
    }
    return answer;
}

File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.*;

class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        List<Integer> answer = new ArrayList<Integer>();
        DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy.MM.dd");
        LocalDate currentDate = LocalDate.parse(today, format);
        LocalDate writeDate = null;
        int idx = 1;
        
        for(String privacy : privacies){
            String wd = privacy.split(" ")[0];
            writeDate = LocalDate.parse(wd, format);
            LocalDate expireDate = null;
            for(String term : terms){
                if(term.split(" ")[0].equals(privacy.split(" ")[1])){
                    int whileMonth = Integer.parseInt(term.split(" ")[1]);
                    expireDate = writeDate.plusMonths(whileMonth);
                    expireDate = expireDate.minusDays(1);
                }
            }
            if(expireDate.isBefore(currentDate)){
                answer.add(idx);
            }
            idx++;
        }
        return answer.stream().mapToInt(i -> i).toArray();
    }
}